File: src/main/java/com/ennoblesoft/demo/service/StudentService.java
package com.ennoblesoft.demo.service;

import com.ennoblesoft.demo.entity.Student;
import com.ennoblesoft.demo.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StudentService {
    @Autowired
    public StudentRepository studentRepository;

    public Student create(Student s) {
        return studentRepository.save(s);
    }

    public List<Student> get(String name) {
        return studentRepository.findByName(name);
    }
}

File: src/main/kotlin/com/ennoblesoft/springdemo/service/StudentService.kt
package com.ennoblesoft.springdemo.service

import com.ennoblesoft.springdemo.entity.Student
import com.ennoblesoft.springdemo.repository.StudentRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class StudentService {
    @Autowired
    var studentRepository: StudentRepository? = null

    fun create(s: Student): Student {
        return studentRepository!!.save(s)
    }

    operator fun get(name: String): List<Student> {
        return studentRepository!!.findByName(name)
    }
}
